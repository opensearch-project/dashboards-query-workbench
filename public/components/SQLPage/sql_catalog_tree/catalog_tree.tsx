/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiComboBoxOptionOption, EuiEmptyPrompt, EuiFlexItem, EuiIcon } from '@elastic/eui';
import React from 'react';
import { OSTree } from './os_tree';
import { S3Tree } from './s3_tree';

interface CatalogTreeProps {
  selectedItems: EuiComboBoxOptionOption[];
  updateSQLQueries: (query: string) => void;
  refreshTree: boolean;
  dataSourceEnabled: boolean;
  dataSourceMDSId: string;
  clusterTab: string;
  language: string;
  updatePPLQueries: (query: string) => void;
}

export const CatalogTree = ({ selectedItems, updateSQLQueries, refreshTree, dataSourceEnabled, dataSourceMDSId, clusterTab, language, updatePPLQueries}: CatalogTreeProps) => {
  return (
    <>
      {selectedItems !== undefined && selectedItems[0].label === 'OpenSearch' && clusterTab !== 'Data source Connections'? (
        <OSTree
          selectedItems={selectedItems}
          updateSQLQueries={updateSQLQueries}
          refreshTree={refreshTree}
          dataSourceEnabled={dataSourceEnabled}
          dataSourceMDSId={dataSourceMDSId}
        />
      )  : (selectedItems[0].label !== 'OpenSearch' && clusterTab === 'Data source Connections')? (
        <S3Tree
          dataSource={selectedItems[0].label}
          updateSQLQueries={updateSQLQueries}
          refreshTree={refreshTree}
          dataSourceEnabled={dataSourceEnabled}
          dataSourceMDSId={dataSourceMDSId}
          language={language}
          updatePPLQueries={updatePPLQueries}
        />
      ):(
        <EuiFlexItem grow={false}>
          <EuiEmptyPrompt
            icon={<EuiIcon type="database" size="m" />}
            iconColor="subdued"
            titleSize="xs"
            body={<p>Select a Data Source Connection to fetch Databases</p>}
          />
        </EuiFlexItem>
      )}
    </>
  );
};
