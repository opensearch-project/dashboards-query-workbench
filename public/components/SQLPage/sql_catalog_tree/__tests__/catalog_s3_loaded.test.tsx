/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { render, waitFor } from '@testing-library/react';
import React from 'react';
import { httpClientMock } from '../../../../../test/mocks';
import { mockDataSelectQuery, mockHttpQuery } from '../../../../../test/mocks/mockData';
import { CatalogTree } from '../catalog_tree';

jest.mock('../../../../dependencies/register_observability_dependencies', () => ({
  getRenderAccelerationDetailsFlyout: jest.fn(() => jest.fn()),
  getRenderCreateAccelerationFlyout: jest.fn(() => jest.fn()),
  setRenderAccelerationDetailsFlyout: jest.fn(() => jest.fn()),
  setRenderCreateAccelerationFlyout: jest.fn(() => jest.fn()),
}));

jest.mock('../../../../framework/catalog_cache_refs', () => ({
  catalogCacheRefs: {
    useLoadDatabasesToCache: jest.fn(() => ({
      loadStatus: '',
      startLoading: jest.fn(),
      stopLoading: jest.fn(),
    })),
    useLoadTablesToCache: jest.fn(() => ({
      loadStatus: '',
      startLoading: jest.fn(),
      stopLoading: jest.fn(),
    })),
    useLoadTableColumnsToCache: jest.fn(() => ({
      loadStatus: '',
      startLoading: jest.fn(),
      stopLoading: jest.fn(),
    })),
    useLoadAccelerationsToCache: jest.fn(() => ({
      loadStatus: '',
      startLoading: jest.fn(),
      stopLoading: jest.fn(),
    })),
    CatalogCacheManager: {
      getOrCreateDataSource: jest.fn(() => ({
        name: 'my_s3',
        lastUpdated: '',
        status: 'Updated',
        databases: [
          {
            name: 'db1',
          },
          {
            name: 'db2',
          },
        ],
      })),
    },
  },
}));

describe('Test Catalog tree', () => {
  it('Loading state S3 catalog tree', async () => {
    const client = httpClientMock;
    client.post = jest.fn().mockResolvedValue(mockHttpQuery);
    client.get = jest.fn().mockResolvedValue(mockDataSelectQuery);
    const { getByText } = render(
      <CatalogTree
        selectedItems={[{ label: 'my_s3' }]}
        updateSQLQueries={jest.fn()}
        refreshTree={true}
        dataSourceEnabled={false} 
        dataSourceMDSId={''} 
        clusterTab={'Data source Connections'} 
        language={'SQL'} 
      />
    );

    await waitFor(() => {
      expect(getByText('db1')).toBeInTheDocument();
    });

    expect(document.body.children[0]).toMatchSnapshot();
  });
});
