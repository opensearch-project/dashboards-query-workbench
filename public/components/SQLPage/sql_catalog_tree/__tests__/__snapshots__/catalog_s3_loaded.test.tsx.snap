// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Catalog tree Loading state S3 catalog tree 1`] = `
<div>
  <div>
    <div
      class="euiText euiText--medium euiTreeView__wrapper"
    >
      <p
        class="euiScreenReaderOnly"
        id="some_html_id--instruction"
      >
        You can quickly navigate this list using arrow keys.
      </p>
      <ul
        aria-describedby="some_html_id--instruction"
        aria-label="S3 Datasource Folder Tree"
        class="euiTreeView workbench-tree"
        data-test-subj="s3-datasource-tree"
        id="some_html_id"
      >
        <li
          class="euiTreeView__node"
        >
          <button
            aria-controls="some_html_id"
            aria-expanded="false"
            class="euiTreeView__nodeInner"
            data-test-subj="euiTreeViewButton-some_html_id"
            id="element_0"
          >
            <span
              class="euiTreeView__iconWrapper"
            >
              <svg
                aria-hidden="true"
                class="euiIcon euiIcon--medium euiIcon-isLoading"
                focusable="false"
                height="16"
                role="img"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                />
              </svg>
            </span>
            <span
              class="euiTreeView__nodeLabel"
            >
              <div>
                <span
                  class="euiToolTipAnchor"
                >
                  <div
                    class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
                  >
                    <div
                      class="euiFlexItem euiFlexItem--flexGrowZero"
                    >
                      <div
                        class="euiText euiText--medium"
                      >
                        db1
                         
                      </div>
                    </div>
                  </div>
                </span>
              </div>
            </span>
          </button>
          <div
            id="some_html_id"
          />
        </li>
        <li
          class="euiTreeView__node"
        >
          <button
            aria-controls="some_html_id"
            aria-expanded="false"
            class="euiTreeView__nodeInner"
            data-test-subj="euiTreeViewButton-some_html_id"
            id="element_1"
          >
            <span
              class="euiTreeView__iconWrapper"
            >
              <svg
                aria-hidden="true"
                class="euiIcon euiIcon--medium euiIcon-isLoading"
                focusable="false"
                height="16"
                role="img"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5.277 10.088c.02.014.04.03.057.047.582.55 1.134.812 1.666.812.586 0 1.84-.293 3.713-.88L9 6.212V2H7v4.212l-1.723 3.876Zm-.438.987L3.539 14h8.922l-1.32-2.969C9.096 11.677 7.733 12 7 12c-.74 0-1.463-.315-2.161-.925ZM6 2H5V1h6v1h-1v4l3.375 7.594A1 1 0 0 1 12.461 15H3.54a1 1 0 0 1-.914-1.406L6 6V2Z"
                />
              </svg>
            </span>
            <span
              class="euiTreeView__nodeLabel"
            >
              <div>
                <span
                  class="euiToolTipAnchor"
                >
                  <div
                    class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
                  >
                    <div
                      class="euiFlexItem euiFlexItem--flexGrowZero"
                    >
                      <div
                        class="euiText euiText--medium"
                      >
                        db2
                         
                      </div>
                    </div>
                  </div>
                </span>
              </div>
            </span>
          </button>
          <div
            id="some_html_id"
          />
        </li>
      </ul>
    </div>
  </div>
</div>
`;
